ReactiveObjC/NSObject+RACSelectorSignal.m

```diff
@@ -75,26 +52,47 @@ static void RACSwizzleForwardInvocation(Class class) {
 	// was no existing implementation, throw an unrecognized selector
 	// exception.
 	id newForwardInvocation = ^(id self, NSInvocation *invocation) {
-		BOOL matched = RACForwardInvocation(self, invocation);
-		if (matched) return;
+		SEL originalSelector = invocation.selector;
+		SEL aliasSelector = RACAliasForSelector(invocation.selector);
+		RACSubject* subject = objc_getAssociatedObject(self, aliasSelector);
+
+		BOOL responseToSelector = [superclass instancesRespondToSelector:originalSelector];
+		BOOL forward = NO;
+		if (responseToSelector) {
+			Method method = class_getInstanceMethod(superclass, invocation.selector);
+			IMP impl = method_getImplementation(method);
+
+			if (impl != _objc_msgForward) {
+				class_replaceMethod(object_getClass(self), aliasSelector, impl, method_getTypeEncoding(method));
```

Comment:
This looks nasty, but there is no reliable way to invoke a specific IMP with `NSInvocation` in the standard toolbox.